homeassistant/components/connector/cover.py

```diff
@@ -0,0 +1,293 @@
+        """Subscribe to multicast pushes."""
+        self._blind.register_callback(self._push_callback)
+        await super().async_added_to_hass()
+
+    async def async_will_remove_from_hass(self):
+        """Unsubscribe when removed."""
+        self._blind.remove_callback()
+        return super().async_will_remove_from_hass()
+
+    @property
+    def current_cover_position(self):
+        """Return the current position."""
+        return 100 - self._blind.position
+
+    def open_cover(self, **kwargs):
+        """Open the cover."""
+        self._blind.open()
+
+    def close_cover(self, **kwargs):
+        """Close cover."""
+        self._blind.close()
+
+    def set_cover_position(self, **kwargs):
+        """Move the cover to a specific position."""
+        position = kwargs[ATTR_POSITION]
+        self._blind.target_position(100 - position)
+
+    def stop_cover(self, **kwargs):
+        """Stop the cover."""
+        self._blind.stop()
+
+
+class TiltDevice(TwoWayDevice):
+    """Representation of a Motion Blind Device."""
+
+    @property
+    def current_cover_tilt_position(self):
+        """Return current angle of cover."""
+        if self._blind.angle is None:
+            return None
+        return self._blind.angle * 100 / 180
+
+    def open_cover_tilt(self, **kwargs):
+        """Open the cover tilt."""
+        self._blind.target_angle(180)
+
+    def close_cover_tilt(self, **kwargs):
+        """Close the cover tilt."""
+        self._blind.target_angle(0)
+
+    def set_cover_tilt_position(self, **kwargs):
+        """Move the cover tilt to a specific position."""
+        angle = kwargs[ATTR_TILT_POSITION] * 180 / 100
+        self._blind.target_angle(angle)
+
+    def stop_cover_tilt(self, **kwargs):
+        """Stop the cover."""
+        self._blind.stop()
+
+
+class OneWayDevice(CoordinatorEntity, CoverEntity):
+    """Representation of a Motion Blind Device."""
+
+    def __init__(self, coordinator, blind, device_class, config_entry):
+        """Initialize the blind."""
+        super().__init__(coordinator)
+
+        self._blind = blind
+        self._device_class = device_class
+        self._config_entry = config_entry
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the blind."""
+        return self._blind.mac
+
+    @property
+    def device_info(self):
+        """Return the device info of the blind."""
+        device_info = {
+            "identifiers": {(DOMAIN, self._blind.mac)},
+            "manufacturer": MANUFACTURER,
+            "name": f"{self._blind.mac}",
+            "model": "Curtain",
+            "via_device": (DOMAIN, self._config_entry.unique_id),
+        }
+
+        return device_info
+
+    @property
+    def name(self):
+        """Return the name of the blind."""
+        return f"{self._blind.mac[-4:]}"
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return True
```

Comment:
Same here

Reply:
Because the Connector integration obtains the motor list through the HUB local UDP interface (the HUB and the motor communicate through 433MHZ), it is impossible to judge whether the motor is available through this interface, so by default, as long as the motor list is obtained, it means that the motor is available